### DISCUSSION TITLE
Need help in using tch-exploit with DGA 4331

### DISCUSSION THREAD

**Discussion by TheHooligan95** [hack-technicolor/hack-technicolor#318] [Q&A] [answered]
Using Ubuntu

I've made a domain that points to 10.42.1.1 I'm running it from my own apache2 based webserver,  which is also the same pc I'm trying to root the DGA4331 with.

Then, I type nm-connection-editor in terminal and set the ipv4 connection to the DGA4331 to shared with other computers, and also create a new VLAN on id 83 also shared with other computers, because the DGA4331 by default listens DHCP request on VLAN 83.

 Then, I start a nginx proxy with this config

`server {
	listen 80 default_server;
	listen [::]:80 default_server;
        listen 443 ssl default_server;
	listen [::]:443 ssl default_server;

	root /var/www/html;

	server_name _;

	location / {
		   proxy_pass http://localhost:4346;
	}

        ssl_certificate      /etc/letsencrypt/live/DOMAIN/fullchain.pem;
        ssl_certificate_key  /etc/letsencrypt/live/DOMAIN/privkey.pem;
        ssl_protocols       TLSv1 TLSv1.1 TLSv1.2;

}`

Then, I finally open tch-exploit with the following command

`sudo ./tch-exploit-linux --acsurl="https://DOMAIN.xyz/" --ip="10.42.1.1" --dhcponly`

Then, I factory reset the router, wait for some stuff to show up on the tch-exploit terminal, and run A SECOND INSTANCE of tch-exploit with the following 
`
sudo ./tch-exploit-linux --port=4346 --ip="10.42.1.1"`

According to more experienced users than me, this should work, but it doesn't. Any ideas?

**Comment by gamecuber6** [comment]
i'm having the same issue, on firmware version 1.2.0. have you found a fix?

**Reply by TheHooligan95** [reply, marked as answer]
https://www.ilpuntotecnico.com/forum/index.php/topic,84353.msg278862.html#msg278862 

Follow this post, step by step. You can translate it if you don't speak italian. After a few tries, the only thing I had to do was waiting something like 5 minutes.

**Reply by gamecuber6** [reply]
thank you, that was the guide i was following, and getting the exact issue you were getting however.
my actual issue was that i didn't set the reverse proxy up properly, which after fixing has made the exploit finally work. ty anyhow!
